packages/desktop-client/src/components/modals/Notes.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+
+import { useLiveQuery } from 'loot-core/src/client/query-hooks';
+import q from 'loot-core/src/shared/query';
+
+import Check from '../../icons/v2/Check';
+import { type CommonModalProps } from '../../types/modals';
+import Button from '../common/Button';
+import Modal from '../common/Modal';
+import View from '../common/View';
+import NotesComponent from '../Notes';
+
+type NotesProps = {
+  modalProps: CommonModalProps;
+  id: string;
+  name: string;
+  onSave: (id: string, notes: string) => void;
+};
+
+export default function Notes({ modalProps, id, name, onSave }: NotesProps) {
+  const data = useLiveQuery(() => q('notes').filter({ id }).select('*'), [id]);
+  const originalNotes = data && data.length > 0 ? data[0].note : null;
+
+  const [notes, setNotes] = useState(originalNotes);
+  useEffect(() => setNotes(originalNotes), [originalNotes]);
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏this has the potential to lead to some interesting situations where the `notes` input updates while the customer is typing. 

Instead of using an effect and watching for the _original_ value change you could just set the new value `onSave`. That would make the UI slightly more stable in the long term. React effects generally lead to a lot of unintended consequences, so I try to avoid them if possible.

Reply:
I added this so that the component picks up the changes to the notes prop and set it the the local state. I tried removing this and the modal is no longer picking up the new notes that is saved in the database returned by the `useLiveQuery`.